app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesDatabase.java

```diff
@@ -769,4 +802,59 @@ private static void validateAccountId(long accountId) {
             throw new IllegalArgumentException("accountId must be greater than 0");
         }
     }
+
+    /**
+     * Get the category if with the given category title
+     * The method does not support fuzzy search.
+     * Because the category title in database is unique, there will not at most one result.
+     * If there is no such category, database will create it if create flag is set.
+     * Otherwise this method will return -1 as default value.
+     *
+     * @param accountId     The user account Id
+     * @param categoryTitle The category title which will be search in the db
+     *
+     * @return -1 if there is no such category else the corresponding id
+     */
+    @NonNull
+    @WorkerThread
+    private Integer getCategoryIdByTitle(long accountId, @NonNull String categoryTitle) {
+        validateAccountId(accountId);
+        SQLiteDatabase db = getReadableDatabase();
+        Cursor cursor = db.query(
+                table_category,
+                new String[]{key_category_id},
+                key_category_title + " = ? AND " + key_category_account_id + " = ? ",
+                new String[]{categoryTitle, String.valueOf(accountId)},
+                key_category_id,
+                null,
+                key_category_id);
```

Comment:
Why order by `key_category_id`? Shouldn't this be `null`?

Reply:
Why is this be null? The column I get is only `key_category_id`. I do not think that this method will crash.

Reply:
It's not about crashing, just about performance.